File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/cache/CacheManage.java
package com.ashlikun.okhttputils.http.cache;

import com.ashlikun.okhttputils.http.HttpUtils;
import com.ashlikun.okhttputils.http.request.HttpRequest;
import com.ashlikun.orm.LiteOrmUtil;
import com.ashlikun.orm.db.assit.WhereBuilder;

import java.util.List;

/**
 * 作者　　: 李坤
 * 创建时间: 2018/5/11 0011　下午 4:06
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：缓存管理器
 */
public class CacheManage {
    public static boolean update(CacheEntity entity) {
        try {
            LiteOrmUtil.get().update(entity);
            return true;
        } catch (Exception e) {
            return false;
        }
    }

    public static boolean delete(CacheEntity entity) {
        try {
            LiteOrmUtil.get().delete(entity);
            return true;
        } catch (Exception e) {
            return false;
        }
    }

    public static boolean delete(String key) {
        try {
            LiteOrmUtil.get().delete(new WhereBuilder(CacheEntity.class).where("key=?", key));
            return true;
        } catch (Exception e) {
            return false;
        }
    }

    public static boolean delete(HttpRequest request) {
        return delete(getCacheKey(request));
    }

    public static void deleteAll() {
        try {
            LiteOrmUtil.get().delete(CacheEntity.class);
        } catch (Exception e) {

        }
    }

    public static CacheEntity queryById(String id) {
        return LiteOrmUtil.get().queryById(id, CacheEntity.class);
    }

    public static CacheEntity queryById(HttpRequest request) {
        return LiteOrmUtil.get().queryById(getCacheKey(request), CacheEntity.class);
    }

    public static List<CacheEntity> queryAll() {
        return LiteOrmUtil.get().query(CacheEntity.class);
    }

    public static String getCacheKey(HttpRequest request) {
        if (request == null) {
            return null;
        }
        if (request.getMethod().equals("GET")) {
            return request.getUrl().toString();
        } else {
            return HttpUtils.createUrlFromParams(request.getUrl(), request.getParams()).toString();
        }
    }
}

File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/cache/CacheManage.kt
package com.ashlikun.okhttputils.http.cache

import android.util.Log
import com.ashlikun.okhttputils.http.HttpUtils
import com.ashlikun.okhttputils.http.cache.CacheEntity
import com.ashlikun.okhttputils.http.request.HttpRequest
import com.ashlikun.orm.LiteOrmUtil
import com.ashlikun.orm.db.assit.WhereBuilder

/**
 * @author　　: 李坤
 * 创建时间: 2021.12.16 13:57
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：缓存管理器
 */
object CacheManage {
    fun update(entity: CacheEntity): Boolean {
        return try {
            LiteOrmUtil.get().update(entity)
            true
        } catch (e: Exception) {
            false
        }
    }

    fun delete(entity: CacheEntity): Boolean {
        return try {
            LiteOrmUtil.get().delete(entity)
            true
        } catch (e: Exception) {
            false
        }
    }

    fun delete(key: String): Boolean {
        return try {
            LiteOrmUtil.get().delete(
                WhereBuilder(
                    CacheEntity::class.java
                ).where("key=?", key)
            )
            true
        } catch (e: Exception) {
            false
        }
    }

    fun delete(request: HttpRequest): Boolean {
        return delete(getCacheKey(request))
    }

    fun deleteAll() {
        try {
            LiteOrmUtil.get().delete(CacheEntity::class.java)
        } catch (e: Exception) {
        }
    }

    fun queryById(id: String): CacheEntity? {
        return LiteOrmUtil.get().queryById(id, CacheEntity::class.java)
    }

    fun queryById(request: HttpRequest): CacheEntity? {
        return LiteOrmUtil.get().queryById(getCacheKey(request), CacheEntity::class.java)
    }

    fun queryAll(): List<CacheEntity> {
        return LiteOrmUtil.get().query(CacheEntity::class.java)
    }

    fun getCacheKey(request: HttpRequest): String {
        return request.cacheKey
    }
}